rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java

```diff
@@ -0,0 +1,1105 @@
+                PrecompiledContracts.BRIDGE_ADDR,
+                bridgeConstants,
+                activations
+        );
+
+        BtcBlockStoreWithCache.Factory btcBlockStoreFactory = new RepositoryBtcBlockStoreWithCache.Factory(
+                bridgeConstants.getBtcParams(),
+                100,
+                100
+        );
+
+        BtcBlockStore btcBlockStore = btcBlockStoreFactory.newInstance(repository, bridgeConstants, bridgeStorageProvider, activations);
+
+        // Setting a chain head different than genesis to avoid having to read the checkpoints file
+        addNewBtcBlockOnTipOfChain(btcBlockStore, bridgeConstants);
+
+        repository.save();
+
+        BridgeEventLogger bridgeEventLogger = mock(BridgeEventLogger.class);
+
+        long blockNumber = 0;
+
+        /***
+         * Creation phase
+         */
+
+        Block initialBlock = mock(Block.class);
+        when(initialBlock.getNumber()).thenReturn(blockNumber);
+
+        BridgeSupport bridgeSupport = new BridgeSupportBuilder()
+                .withProvider(bridgeStorageProvider)
+                .withRepository(repository)
+                .withEventLogger(bridgeEventLogger)
+                .withExecutionBlock(initialBlock)
+                .withActivations(activations)
+                .withBridgeConstants(bridgeConstants)
+                .withBtcBlockStoreFactory(btcBlockStoreFactory)
+                .withPeginInstructionsProvider(new PeginInstructionsProvider())
+                .build();
+
+        List<FederationMember> originalPowpegMembers = keysFrom.stream().map(theseKeys -> new FederationMember(theseKeys.getLeft(), theseKeys.getMiddle(), theseKeys.getRight())).collect(Collectors.toList());
+
+        Federation originalPowpeg;
+        switch (federationTypeFrom) {
+            case erp:
+                originalPowpeg = new ErpFederation(
+                        originalPowpegMembers,
+                        Instant.now(),
+                        0,
+                        bridgeConstants.getBtcParams(),
+                        bridgeConstants.getErpFedPubKeysList(),
+                        bridgeConstants.getErpFedActivationDelay(),
+                        activations
+                );
+                break;
+            case p2sh:
+                originalPowpeg = new P2shErpFederation(
+                        originalPowpegMembers,
+                        Instant.now(),
+                        0,
+                        bridgeConstants.getBtcParams(),
+                        bridgeConstants.getErpFedPubKeysList(),
+                        bridgeConstants.getErpFedActivationDelay(),
+                        activations
+                );
+                // TODO: CHECK REDEEMSCRIPT
+                break;
+            default:
+                throw new Exception();
+        }
+
+        Assert.assertEquals(expectedAddressFrom, originalPowpeg.getAddress());
+
+        // Set original powpeg informatino
```

Comment:
```suggestion
        // Set original powpeg information
```

Reply:
informatino is like information but for latinos 🤡 
I will fix it 😄